src/main/scala/net/cakesolutions/CakeBuildInfoPlugin.scala

```diff
@@ -25,9 +33,54 @@ object CakeBuildInfoPlugin extends AutoPlugin {
         Try("git rev-parse --verify HEAD".!! dropRight 1) getOrElse "n/a"
       )
     ),
-    buildInfoPackage := s"${organization.value}.${(name in ThisBuild).value}.build",
+    buildInfoPackage :=
+      s"${organization.value}.${(name in ThisBuild).value}.build",
     buildInfoOptions += BuildInfoOption.BuildTime,
     buildInfoOptions += BuildInfoOption.ToJson
   )
 
 }
+
+/**
+  * Build keys that will be auto-imported when this plugin is enabled.
+  */
+object CakeBuildInfoKeys {
+  /**
+    * Map holding generic set of labelled values. These are use to label jar
+    * manifests and Docker containers.
+    */
+  val generalInfo: Map[String, String] = Map(
+    "buildDatetime" -> buildDatetime,
+    "buildHost" -> buildHost,
+    "buildUser" -> buildUser,
+    "gitBranch" -> gitBranch,
+    "gitCommitHash" -> gitCommitHash,
+    "gitRepository" -> gitRepository
+  )
+
+  /**
+    * Map holding Docker specific labelled values. These are used to label
+    * Docker containers.
+    */
+  val dockerInfo: Map[String, String] = Map(
+    "buildDockerVersion" -> buildDockerVersion
+  )
+
+  private def buildDatetime =
+    ZonedDateTime.now(Clock.systemUTC()).format(DateTimeFormatter.ISO_INSTANT)
+
+  private def buildDockerVersion = "docker --version".!!.trim
```

Comment:
And that's why we need to test for their existence. Where possible, we should also validate the output they generate.

I'd suggest that trying to make this code cross platform compatible is beyond the scope of this task. Simply being able to detect command existence should be sufficient for now.